python_modules/dagster/dagster/_core/snap/node.py

```diff
@@ -176,15 +177,39 @@ def get_output_snap(self, name: str) -> OutputDefSnap:
 
 
 @whitelist_for_serdes(storage_name="SolidDefSnap")
-@record
-class OpDefSnap:
+@record_custom
```

Comment:
why does this now need to be custom? does the backcompat not work if pool is just `pool: Optional[str] = None`?